src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/BuildPickUpMessage.java

```diff
@@ -0,0 +1,62 @@
+package com.minecolonies.coremod.network.messages.server.colony.building;
+
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.colony.buildings.views.IBuildingView;
+import com.minecolonies.coremod.network.messages.server.AbstractBuildingServerMessage;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.network.NetworkEvent;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Picks up the building block with the level.
+ */
+public class BuildPickUpMessage extends AbstractBuildingServerMessage<IBuilding>
+{
+    /**
+     * Empty constructor used when registering the
+     */
+    public BuildPickUpMessage()
+    {
+        super();
+    }
+
+    /**
+     * Creates a build request
+     *
+     * @param building the building we're executing on.
+     */
+    public BuildPickUpMessage(@NotNull final IBuildingView building)
+    {
+        super(building);
+    }
+
+    @Override
+    public void fromBytesOverride(@NotNull final PacketBuffer buf)
+    {
+
+    }
+
+    @Override
+    public void toBytesOverride(@NotNull final PacketBuffer buf)
+    {
+
+    }
+
+    @Override
+    public void onExecute(final NetworkEvent.Context ctxIn, final boolean isLogicalServer, final IColony colony, final IBuilding building)
+    {
+        final PlayerEntity player = ctxIn.getSender();
+        building.pickUp(player);
+        for (final BlockPos childPos : building.getChildren())
```

Comment:
we got some buildings of which you arent supposed to obtain a hutblock

Reply:
actually this might be a bad idea anyways, those blocks are integrated into the main buildings and are never meant to be seperated as having seperate blocks with level would do

Reply:
kk